src/main/java/org/gridsuite/network/map/NetworkMapService.java

```diff
@@ -707,4 +722,23 @@ public AllMapData getAll(UUID networkUuid, String variantId, List<String> substa
             return res.stream().collect(Collectors.toList());
         }
     }
+
+    public List<VoltageLevelMapData> getVoltageLevels(UUID networkUuid, String variantId, List<String> substationsId) {
```

Comment:
@FranckLecuyer how this method is used? Mostly with one substation id when list is not null? In that case your impl is ok (I mean performance point of view) but if substationIds become to be a little bit bigger (tens of), you are going to have performance issues and you should use COLLECTION instead.

Reply:
Actually, substationsId is always null when calling this method, but i kept this parameter for homogenizatioin purpose with other existing methods ...